### PR TITLE
Update Dependencies Issue #73

### PR DESCRIPTION
**Pull request by nelsonic**
This is a _maintenance_ PR, it _should_ preserve backward compatibility with `v2.2.2`.
I just don't like seeing 🔴 badges in projects ... #73 

What's included? 
+ [x] Updates the _main_ dependency `hoek` to `@hapi/hoek` (_Hapi's new namespacing convention..._) 🙄 
+ [x] Updates all `devDependencies` to latest versions 
  + [x] including migrating the ones that _work_ without modification to the `@hapi/{module}` namespace.
+ [x] Update badges to `flat-square` for consistency with our other projects.

@iteles please review/merge when you can. Thanks! ✨

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+ * isFunction checks if a given value is a function.
+ * @param {Object} functionToCheck - the object we want to confirm is a function
+ * @returns {Boolean} true|false
+ */
 function isFunction(functionToCheck) {
-  return functionToCheck && {}.toString.call(functionToCheck) === '[object Function]';
+  const toString = Object.prototype.toString;
+  return functionToCheck
+    && toString.call(functionToCheck) === '[object Function]'
+    || toString.call(functionToCheck) === '[object AsyncFunction]';
```
Comment (Hydrock) [inline lib/index.js:15]: 💪

File: lib/index.js [hunk 2]
```diff
@@ -114,9 +122,10 @@ exports.plugin = {
           if (currentCodeConfig && currentCodeConfig.redirect) {
             // if redirect is function invoke it with the request object
             if (isFunction(currentCodeConfig.redirect)) {
-              return reply.redirect(currentCodeConfig.redirect(request));
-            } 
-            else { 
+              const url = currentCodeConfig.redirect(request)
+              return reply.redirect(url);
```
Comment (Hydrock) [inline lib/index.js:126]: yes - it is easier to read

File: test/redirect_server_example.js [hunk 3]
```diff
@@ -2,15 +2,16 @@
...
+var Hoek = require('@hapi/hoek');
 
 var config = {
   "401": { // if the statusCode is 401 redirect to /login page/endpoint
     "redirect": "/login"
   },
   "403": {
     "redirect": function (request) {
-      return "/login?redirect=" + request.url.path
+      var redirectString = request.url.pathname + request.url.search;
```
Comment (Hydrock) [inline test/redirect_server_example.js:13]: just wanted to do this 👍

**Review by Hydrock — APPROVED** [review]
It's looks good
please, check 227 line in README file

it's about string `return "/login?redirect=" + request.url.path`

url.path - is incorrect

**Comment by nelsonic** [comment]
@Hydrock thanks for your review. Well spotted on the `request.url.path` > `request.url.pathname` change in the `REAMDE.md#227` 👍  (_I've made the change suggested_) 

@iteles this should be good to review/merge. (_thanks!_)

File: test/redirect_server_example.js (hunk 3 above)
Comment (nelsonic) [reply test/redirect_server_example.js:13]: I just copied over the code you wrote in #70 to make the test/example server consistent.
(_the test was failing on my `localhost` when I was trying to update the dependencies ..._)

**Comment by Hydrock** [comment]
> @Hydrock thanks for your review. Well spotted on the `request.url.path` > `request.url.pathname` change in the `REAMDE.md#227` 👍 (_I've made the change suggested_)
> 
> @iteles this should be good to review/merge. (_thanks!_)
yes, i saw. I think it's good idea

File: lib/index.js (hunk 2 above)
Comment (nelsonic) [reply lib/index.js:126]: I was debugging why redirect wasn't working in the test see below ⬇️ https://github.com/dwyl/hapi-error/pull/74/files/405fa01cfcffbaf8f37812f3b75a4f1947010a04#r372028342
and just decided to leave in this temporary variable. V8 will garbage collect it pretty quickly or might even in-line it in the JIT so I'm not concerned with perf. 👍

File: lib/index.js (hunk 1 above)
Comment (nelsonic) [reply lib/index.js:15]: Borrowed from [dwyl/hapi-auth-jwt2/lib/index.js#L56](https://github.com/dwyl/hapi-auth-jwt2/blob/490f2fe7761818e5cb118c8cfe44b019cf7d35b4/lib/index.js#L56) not worth creating/using an NPM module for this but worth documenting it with a JSDOC comment. 😉

File: lib/index.js [hunk 4]
```diff
@@ -96,14 +104,14 @@ exports.plugin = {
             method: req.method, // e.g GET/POST
             url: request.url.path, // the path the person requested
             headers: request.raw.req.headers, // all HTTP Headers
-            info: request.info, // all additional request info (useful for debug)
-            auth: request.auth, // any authentication details e.g. the decoded JWT
+            info: request.info, // all additional request info (useful to debug)
```
Comment (iteles) [inline lib/index.js:107]:
```suggestion
            info: request.info, // all additional request info for debugging
```

Comment (iteles) [reply lib/index.js:107]: I know this goes onto the next line but does make more sense in English and doesn't skip to the next line in the rich diff.

File: lib/index.js (hunk 1 above)
Comment (Hydrock) [reply lib/index.js:15]: I didn't know about this option, or forgot - [object AsyncFunction]

**Review by iteles — APPROVED** [review]
🙌 Thanks @nelsonic, a couple of readability improvements too 👍

File: lib/index.js (hunk 4 above)
Comment (nelsonic) [reply lib/index.js:107]: requires horizontal scrolling  in my code editor ... I'd prefer to keep it to 80 characters for consistency.  I'd prefer to remove the `(useful to debug)` _completely_ than go over the 80 char limit.

Comment (iteles) [reply lib/index.js:107]: Hm, do you want to change it back before I merge?

File: lib/index.js [hunk 5]
```diff
@@ -96,14 +104,14 @@ exports.plugin = {
             method: req.method, // e.g GET/POST
             url: request.url.path, // the path the person requested
             headers: request.raw.req.headers, // all HTTP Headers
-            info: request.info, // all additional request info (useful for debug)
-            auth: request.auth, // any authentication details e.g. the decoded JWT
+            info: request.info, // all additional request info (useful for debugging)
```
Comment (iteles) [inline lib/index.js]:
```suggestion
            info: request.info, // all additional request info for debugging
```

**Pull request merged**